### ISSUE TITLE
Update to OJS Version 3.4

### ISSUE THREAD

**Issue by jpmmartins** [ulsdevteam/inlineHtmlGalley#10]
We are using this plugin in an OJS 3.3 installation to show the html content in the main page of a particular article. Are there any plans to update the plugin?

**Comment by ctgraham** [comment]
Yes, we have a current project upgrading each of our plugins to support 3.4.

**Comment by jbogdani** [comment]
Hello,
I had the plugin installed in a 3.3.x OJS instance, which was updated to the 3.4.x version by the company that manages our journal for ‘security reasons’. The update broke our journal and I am wondering if it would be easier to roll back or to upgrade the plugin.
Do you have a timeline for the new version? The sysadmins are not willing to downgrade since they are hosting many journals on the same instance...
Maybe I can also help with developing/testing if necessary.

Thank you for your work.

**Comment by trp89** [comment]
@jbogdani I just pushed changes for the 3.4 version of this plugin. Above I have references to two commits containing those specific changes. Let me know how the changes look.

**Comment by jbogdani** [comment]
@trp89 thank you.
Sorry for the late replay, but it took a while to get the update applied to the (managed) OJS instance.

Everything seems to work fine, many thanks for your work!

**Comment by jpmmartins** [comment]
I am having troubles loading the DOI on the sidebar. It is activated in Website Settings>Appearance>Setup>Sidebar. I believe it might have to do with the rewrite of the function `lib/pkp/classes/plugins/PluginRegistry.php:107` in version 3.4.

OJS 3.3: https://journals.ub.uni-koeln.de/index.php/phidi/article/view/2412

![image](https://github.com/user-attachments/assets/6f5474e5-9eb2-4774-be0b-c39eb0419c27)

Should I open a specific issue, or should this be a running issue for fixing problems with the update?

Nice to meet you at the PKP Sprint @ctgraham.

**Comment by ctgraham** [comment]
Hi, @jpmmartins , this issue is a good place to report bugs in the draft 3.4 implementation.

@trp89 , @jpmmartins  suggests looking at `PluginRegistry::loadCategory()`
https://github.com/pkp/pkp-lib/blob/bbdfc8a66734904f1b7f57f847838c0699b77e44/classes/plugins/PluginRegistry.php#L107

For the DOI to be presented, we'll need the DOI PubIdPlugin to be loaded into the template manager at the time of block display.  ( The block display looks for it here: 
https://github.com/ulsdevteam/inlineHtmlGalley/blob/f093357d259fed076404a4a36daa31f9483035cb/InlineHtmlGalleySidebarBlockPlugin.php#L68-L82 )

**Comment by jpmmartins** [comment]
The reason why it is not working seems to be because the `plugins/pubIds/doi` does not exist anymore.

@colberg found a workaround for the bootstrap3 theme where Alec suggested going via the article data.

https://github.com/pkp/bootstrap3/commit/0a129341e6a216a6a71b51a17609dc2e855cbfde

I have looked into the plugin and the solution seems not be just changing `plugins/generic/inlineHtmlGalley/templates/blockDoi.tpl` but something needs to be changed also on in the `function getContents` of [InlineHtmlGalleySidebarBlockPlugin.php](https://github.com/ulsdevteam/inlineHtmlGalley/blob/f093357d259fed076404a4a36daa31f9483035cb/InlineHtmlGalleySidebarBlockPlugin.php#L68-L82)

**Comment by jpmmartins** [comment]
I was installing the version of the plugin on a ojs-3.4.0-8 instance, and I also got an error related to the the new carbon class. It will not throw an error anymore after correcting the line https://github.com/ulsdevteam/inlineHtmlGalley/blob/5af7a008c34eae779212ab892a9a60c5ef4a1c58/templates/blockPublishedDate.tpl#L21

I will do a pull request for you to analyse.

**Comment by jpmmartins** [comment]
I have created a pull request to fix the date formating and the doi block. 

https://github.com/ulsdevteam/inlineHtmlGalley/pull/16

I wonder why is there a duplication in the doi.readerDisplayName both in the header of the box and in the listing of the values for both Date Published and DOI. 

I also tried the plugin on Version 3.5 but there are corrections to be done. Maybe we can open another issue to port it.
